checker/tests/nullness/Issue3443.java

```diff
@@ -0,0 +1,18 @@
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+class Issue3443 {
+    static <T extends Supplier3443<@Nullable String>> Supplier3443<String> passThrough(T t) {
+        // :: error: (return.type.incompatible)
+        return t;
+    }
+
+    public static void main(String[] args) {
+        Supplier3443<@Nullable String> s1 = () -> null;
+        Supplier3443<String> s2 = passThrough(s1);
```

Comment:
Shouldn't there also be an error on this line? Either that the inferred type argument is incompatible or that the argument is incompatible?
I guess that is masked by #979 work-arounds, but it might be good to highlight that we would expect an error here.
This is important if `passThrough` is in a library and we don't "see" that there is an error in the implementation.

Reply:
No, the return type of pass through is `Supplier3443<String>`.

Reply:
Yes, I added a comment.